build-tools/config/webpack/webpack.prod.conf.js

```diff
@@ -0,0 +1,165 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const CopyWebpackPlugin = require('copy-webpack-plugin');
+const WebpackCleanupPlugin = require('webpack-cleanup-plugin');
+const OptimizeCssAssetsPlugin = require('optimize-css-assets-webpack-plugin');
+const webpackHelpers = require('./webpackHelpers');
+const BundleAnalyzerPlugin = require('webpack-bundle-analyzer').BundleAnalyzerPlugin;
+const ImageminPlugin = require('imagemin-webpack-plugin').default;
+const FaviconsWebpackPlugin = require('favicons-webpack-plugin');
+
+const env = config.build.env;
+
+const webpackConfig = merge(baseWebpackConfig, {
+  module: {
+    rules: [
+      {
+        test: /\.scss$/,
+        loader: ExtractTextPlugin.extract({
+          use: webpackHelpers.getScssLoaderConfig(),
+        }),
+      },
+      {
+        test: /\.vue$/,
+        use: [webpackHelpers.getVueLoaderConfig(false, config.build.enableESLintLoader)],
+      },
+      {
+        test: /\.(png|jpe?g|gif)(\?.*)?$/,
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              limit: 10000,
+              name: path.posix.join(config.build.versionPath, 'image/[name].[hash:7].[ext]'),
+            },
+          },
+        ],
+      },
+      {
+        test: /\.(woff2?|eot|ttf|otf)(\?.*)?$/,
+        use: [
+          {
+            loader: 'url-loader',
+            options: {
+              limit: 10000,
+              name: path.posix.join(config.build.versionPath, 'font/[name].[hash:7].[ext]'),
+            },
+          },
+        ],
+      },
+    ],
+  },
+  devtool: false,
+  output: {
+    filename: path.posix.join('', config.build.versionPath + '/js/[name].js'),
+    chunkFilename: path.posix.join('', config.build.versionPath + '/js/[id].js'),
+    publicPath: config.build.publicPath,
+  },
+  plugins: [
+    new WebpackCleanupPlugin(),
+    new webpack.optimize.ModuleConcatenationPlugin(),
+    new webpack.DefinePlugin({
+      'process.env': env,
+    }),
+    new webpack.NamedChunksPlugin(),
+    new webpack.optimize.UglifyJsPlugin({
+      compress: {
+        warnings: false,
+      },
+      output: {
+        comments: false,
+      },
+    }),
+    new OptimizeCssAssetsPlugin({
+      cssProcessorOptions: {
+        safe: true,
+      },
+    }),
+    new ExtractTextPlugin({
+      filename: path.posix.join(config.build.versionPath, '/css/[name].css'),
+    }),
+    new HtmlWebpackPlugin({
+      filename: config.build.index,
+      template: 'index.html',
+      inject: true,
+      version: config.build.versionPath,
+      minify: {
+        removeComments: true,
+        collapseWhitespace: true,
+        removeAttributeQuotes: false,
+      },
+      chunksSortMode: 'dependency',
+    }),
+    new ImageminPlugin({
+      disable: !config.build.enableImageOptimization,
+      svgo: null,
+      gifsicle: null,
+      pngquant: config.build.enablePNGQuant ? { quality: '65' } : null,
```

Comment:
Why is quality set to 65 is this standard value ? Shouldn't this come from a config?